In [2]:
import pandas as pd
import numpy as np

In [3]:
compare_output = pd.read_csv('data56_related_files/inStrain_output/instrain_compare_genomeWide_compare.tsv', sep = '\t')
ref_genomes = pd.read_csv('data56_related_files/present_ref_genomes.tsv', sep = '\t', header=None)

In [4]:
compare_output

genome                  name1  \
0       GCA_000615245.1_ASM61524v1_genomic.fna  ERR2241647.sorted.bam   
1       GCA_000615245.1_ASM61524v1_genomic.fna  ERR2241647.sorted.bam   
2       GCA_000615245.1_ASM61524v1_genomic.fna  ERR2241647.sorted.bam   
3       GCA_000615245.1_ASM61524v1_genomic.fna  ERR2241647.sorted.bam   
4       GCA_000615245.1_ASM61524v1_genomic.fna  ERR2241651.sorted.bam   
...                                        ...                    ...   
316036                             S9C3913.fna  SRR9113706.sorted.bam   
316037                             S9C3913.fna  SRR9113706.sorted.bam   
316038                             S9C3913.fna  SRR9113707.sorted.bam   
316039                             S9C3913.fna  SRR9113707.sorted.bam   
316040                             S9C3913.fna  SRR9113725.sorted.bam   

                        name2  coverage_overlap  compared_bases_count  \
0       ERR2241890.sorted.bam               NaN                     0   
1       ERR2241893.sorted.bam               NaN                     0   
2       ERR2241899.sorted.bam               NaN                     0   
3       ERR2241919.sorted.bam               NaN                     0   
4       ERR2241891.sorted.bam               NaN                     0   
...                       ...               ...                   ...   
316036  SRR9113707.sorted.bam               NaN                     0   
316037  SRR9113737.sorted.bam               NaN                     0   
316038  SRR9113725.sorted.bam               NaN                     0   
316039  SRR9113737.sorted.bam           0.14364                  1632   
316040  SRR9113737.sorted.bam               NaN                     0   

        consensus_SNPs  population_SNPs    popANI    conANI  percent_compared  
0                    0                0       NaN       NaN          0.000000  
1                    0                0       NaN       NaN          0.000000  
2                    0                0       NaN       NaN          0.000000  
3                    0                0       NaN       NaN          0.000000  
4                    0                0       NaN       NaN          0.000000  
...                ...              ...       ...       ...               ...  
316036               0                0       NaN       NaN          0.000000  
316037               0                0       NaN       NaN          0.000000  
316038               0                0       NaN       NaN          0.000000  
316039               8                3  0.998162  0.995098          0.001306  
316040               0                0       NaN       NaN          0.000000  

[316041 rows x 10 columns]

In [5]:
compare_out = compare_output[compare_output['coverage_overlap'].notna()]

In [6]:
compare_out['genome'].unique()

array(['GCA_001311425.1_ASM131142v1_genomic.fna',
       'GCA_001311745.1_ASM131174v1_genomic.fna',
       'GCA_001311765.1_ASM131176v1_genomic.fna',
       'GCA_001435375.1_ASM143537v1_genomic.fna',
       'GCA_002299975.1_ASM229997v1_genomic.fna',
       'GCA_002418005.1_ASM241800v1_genomic.fna',
       'GCA_002418055.1_ASM241805v1_genomic.fna',
       'GCA_003666485.1_ASM366648v1_genomic.fna',
       'GCA_004556255.1_ASM455625v1_genomic.fna',
       'GCA_009809575.1_ASM980957v1_genomic.fna',
       'GCA_014651235.1_ASM1465123v1_genomic.fna',
       'GCA_900112995.1_IMG-taxon_2597490368_annotated_assembly_genomic.fna',
       'GCA_900554745.1_UMGS1753_genomic.fna',
       'GCA_900557215.1_UMGS2022_genomic.fna',
       'GCF_000014425.1_ASM1442v1_genomic.fna',
       'GCF_000014445.1_ASM1444v1_genomic.fna',
       'GCF_000016825.1_ASM1682v1_genomic.fna',
       'GCF_000092505.1_ASM9250v1_genomic.fna',
       'GCF_000146325.1_ASM14632v1_genomic.fna',
       'GCF_000159215.1_ASM15921v1_g

In [7]:
compare_present_refs = compare_out[compare_out['genome'].isin(ref_genomes[0])]

In [8]:
compare_present_refs.head(5)

genome                  name1  \
10883  GCA_900557215.1_UMGS2022_genomic.fna  ERR2241647.sorted.bam   
10884  GCA_900557215.1_UMGS2022_genomic.fna  ERR2241647.sorted.bam   
10885  GCA_900557215.1_UMGS2022_genomic.fna  ERR2241647.sorted.bam   
10886  GCA_900557215.1_UMGS2022_genomic.fna  ERR2241647.sorted.bam   
10887  GCA_900557215.1_UMGS2022_genomic.fna  ERR2241647.sorted.bam   

                       name2  coverage_overlap  compared_bases_count  \
10883  ERR2241649.sorted.bam          0.544683                514338   
10884  ERR2241651.sorted.bam          0.337334                 11614   
10885  ERR2241653.sorted.bam          0.531747                 23313   
10886  ERR2241655.sorted.bam          0.654703                 30639   
10887  ERR2241848.sorted.bam          0.389301                 22893   

       consensus_SNPs  population_SNPs    popANI    conANI  percent_compared  
10883            2171              585  0.998863  0.995779          0.430197  
10884              89               44  0.996211  0.992337          0.009714  
10885             163               52  0.997769  0.993008          0.019499  
10886             197               56  0.998172  0.993570          0.025627  
10887             284              150  0.993448  0.987594          0.019148

In [9]:
compare_present_coverage = compare_present_refs[compare_present_refs['percent_compared']>= 0.50]

In [10]:
compare_present_coverage.head(5)

genome                  name1  \
10946    GCA_900557215.1_UMGS2022_genomic.fna  ERR2241649.sorted.bam   
10951    GCA_900557215.1_UMGS2022_genomic.fna  ERR2241649.sorted.bam   
11347    GCA_900557215.1_UMGS2022_genomic.fna  ERR2241860.sorted.bam   
11392    GCA_900557215.1_UMGS2022_genomic.fna  ERR2241887.sorted.bam   
21337  GCF_000159355.1_ASM15935v1_genomic.fna  ERR2241649.sorted.bam   

                       name2  coverage_overlap  compared_bases_count  \
10946  ERR2241887.sorted.bam          0.670722                634580   
10951  ERR2241919.sorted.bam          0.909787                946111   
11347  ERR2241919.sorted.bam          0.596919                603297   
11392  ERR2241919.sorted.bam          0.678538                655768   
21337  ERR2241848.sorted.bam          0.811104               1094802   

       consensus_SNPs  population_SNPs    popANI    conANI  percent_compared  
10946            7866             5305  0.991640  0.987604          0.530769  
10951            6172             1395  0.998526  0.993476          0.791337  
11347            1623              520  0.999138  0.997310          0.504604  
11392           11647            11455  0.982532  0.982239          0.548491  
21337            5161             1005  0.999082  0.995286          0.616518

In [24]:
compare_popANI = compare_present_coverage[compare_present_coverage['popANI'] >= 0.99999]

In [25]:
compare_popANI

genome                  name1  \
129169  GCF_001436315.1_ASM143631v1_genomic.fna  SRR6323197.sorted.bam   
129174  GCF_001436315.1_ASM143631v1_genomic.fna  SRR6323197.sorted.bam   
129498  GCF_001436315.1_ASM143631v1_genomic.fna  SRR6323350.sorted.bam   
129501  GCF_001436315.1_ASM143631v1_genomic.fna  SRR6323350.sorted.bam   
129503  GCF_001436315.1_ASM143631v1_genomic.fna  SRR6323350.sorted.bam   
129533  GCF_001436315.1_ASM143631v1_genomic.fna  SRR6323384.sorted.bam   
129547  GCF_001436315.1_ASM143631v1_genomic.fna  SRR6323414.sorted.bam   
129592  GCF_001436315.1_ASM143631v1_genomic.fna  SRR6323503.sorted.bam   
129618  GCF_001436315.1_ASM143631v1_genomic.fna  SRR6323530.sorted.bam   
285593                             S13C2782.fna  SRR6323350.sorted.bam   
300030                             S28C2674.fna  SRR6323243.sorted.bam   
301964                               S2C205.fna  ERR2241647.sorted.bam   
306145                             S33C4056.fna  SRR6323252.sorted.bam   
306146                             S33C4056.fna  SRR6323252.sorted.bam   
306284                             S33C4056.fna  SRR6323384.sorted.bam   
309196                              S42C891.fna  SRR6323252.sorted.bam   
309913                             S49C1120.fna  SRR6323197.sorted.bam   
309921                             S49C1120.fna  SRR6323197.sorted.bam   
310053                             S49C1120.fna  SRR6323350.sorted.bam   
313036                              S52C595.fna  SRR9113691.sorted.bam   

                        name2  coverage_overlap  compared_bases_count  \
129169  SRR6323503.sorted.bam          0.761489                947065   
129174  SRR6323556.sorted.bam          0.700873                865638   
129498  SRR6323503.sorted.bam          0.753598                944080   
129501  SRR6323530.sorted.bam          0.953995               1251946   
129503  SRR6323556.sorted.bam          0.694339                863226   
129533  SRR6323550.sorted.bam          0.661917                886472   
129547  SRR6323550.sorted.bam          0.673511                885441   
129592  SRR6323530.sorted.bam          0.734300                946174   
129618  SRR6323556.sorted.bam          0.676072                864562   
285593  SRR6323530.sorted.bam          0.924232                854077   
300030  SRR6323367.sorted.bam          0.935632               1274452   
301964  ERR2241649.sorted.bam          0.999385               1613987   
306145  SRR6323384.sorted.bam          0.998241               1334879   
306146  SRR6323414.sorted.bam          0.998191               1335387   
306284  SRR6323414.sorted.bam          0.998677               1335791   
309196  SRR6323414.sorted.bam          0.996205               1761058   
309913  SRR6323350.sorted.bam          0.781665               1424123   
309921  SRR6323530.sorted.bam          0.845559               1648707   
310053  SRR6323530.sorted.bam          0.778136               1466920   
313036  SRR9113737.sorted.bam          0.976907               1675520   

        consensus_SNPs  population_SNPs    popANI    conANI  percent_compared  
129169             955                0  1.000000  0.998992          0.563553  
129174             750                0  1.000000  0.999134          0.515099  
129498            1185                7  0.999993  0.998745          0.561776  
129501            1338                5  0.999996  0.998931          0.744973  
129503             912                3  0.999997  0.998943          0.513664  
129533            2198                8  0.999991  0.997521          0.527497  
129547            4492                6  0.999993  0.994927          0.526883  
129592             903                1  0.999999  0.999046          0.563023  
129618             691                0  1.000000  0.999201          0.514459  
285593            1121                2  0.999998  0.998687          0.630884  
300030               8                5  0.999996  0.999994          0.9

20 strains have been found to be identical between two samples, with ≥50% coverage between the two samples and with a popANI ≥99.999%. 